develop/plugin_tutorial.rst

```diff
@@ -158,402 +104,358 @@ special note of the section that looks like this::
+            self.rng = random.Random()  # create our rng
+            self.rng.seed() # automatically seeds with current time
+
+For this sample plugin, we define a custom constructor (``__init__``) that
+instantiates a random number generator instance and pre-seeds it. This isn't
+technically necessary for Python's ``random`` module, but it helps outline
+how to write a similar constructor. Notice in particular how you must pass in
+an ``irc`` argument in addition to ``self``.
+
+.. warning::
+    Because Limnoria is a multi-network bot, you should generally ignore
+    the ``irc`` instance passed to the plugin constructor.
+    On a manual ``load`` call to a live bot, this will be set to the network
+    the command was run on, but on bot startup, ``irc`` will be (arbitrarily)
+    set to the first network that the bot decides to connect to.
+
+Basic command handler
+---------------------
+
+Our first command definition can immediately follow:
+
+..
+    note: turn off automatic dedent so that the functions appear at the right
+    indent level relative to the class definition
+
+.. code-block::
+    :dedent: 0
+
+        @wrap
+        def random(self, irc, msg, args):
+            """takes no arguments
+
+            Returns the next random number from the random number generator.
+            """
+            irc.reply(str(self.rng.random()))
+
+.. note::
+    All functions used as commands must have an all lowercase name.
+
+A command function taking in no arguments from IRC will still require 4
+arguments; they are as follows:
+
+- ``self``: refers to the class instance. It is common to keep local state
+  for the plugin as instance variables within the plugin class.
+- ``irc``: refers to the IRC network instance the command was called on
+- ``msg``: a :class:`supybot.ircmsgs.IrcMsg` instance; refers to the
+  IRC message that triggered this command.
+- ``args``: a raw list of remaining unconverted arguments; new plugins that
+  use :ref:`@wrap <using-wrap>` for automatic argument type conversion should
+  never need to interact with ``args`` directly.
+
+The function docstring is expected to be in a particular format. First, the very
+first line dictates the argument list to be displayed when someone calls the
+``help`` command on this command (i.e., ``help random``). Then, leave a blank
+line and start the actual help string for the function. Don't worry about the
+fact that it's tabbed in or anything like that, as the help command normalizes
+it to make it look nice. This part should be fairly brief but sufficient to
+explain the function and what (if any) arguments it requires. Remember that this
+should fit in one IRC message which is typically around a 450 character limit.
+
+The :py:meth:`irc.reply <supybot.callbacks.NestedCommandsIrcProxy.reply>` call
+is a bit of magic: it issues a reply the same place as the message that
+triggered the command. i.e. this may be in a channel or in a private
+conversation with the bot.
```

Comment:
```suggestion
The :py:meth:`irc.reply <supybot.callbacks.ReplyIrcProxy.reply>` call
is a bit of magic: it issues a reply the same place as the message that
triggered the command. i.e. this may be in a channel or in a private
conversation with the bot.
```

`ReplyIrcProxy` is the parent class; and IIRC `NestedCommandsIrcProxy` is only passed to nested commands. Regardless, `ReplyIrcProxy`'s `reply` method is better documented